### ISSUE TITLE
[Issue]: CK stage2 fp4 MoE kernel returns non-finite output (nan/inf) for M<=4 on gfx950 (a4w4 per_1x32)

### ISSUE THREAD

**Issue by jhinpan** [ROCm/aiter#3964]
## Summary

aiter's **CK stage2 fp4 MoE kernel** (`moe_ck2stages_gemm2_64x32x32x128_..._FP4X2_FP4X2_B16`,
`QuantType.per_1x32`) returns **non-finite output** (`nan` + `inf`, absmax = fp32 max
`3.4028e38`) for **very small token counts (M ≤ 4)** on gfx950 / MI350X, while the
torch fp32 reference for the exact same inputs is **finite**.

The failure is sharp and deterministic: **M ≤ 4 → non-finite, M ≥ 8 → clean.** It
reproduces for both DeepSeek-V3 and Kimi-K2 a4w4 (afp4/wfp4) MoE shapes. Because
aiter buckets MoE kernels by `get_padded_M(M) = nextPow2(M)`, decode batch sizes
1–4 all land in the affected buckets, so low-concurrency a4w4 MoE decode produces
corrupted (NaN/Inf) outputs.

This is the **stage2** CK kernel, and the failure mode is **non-finite output**
(distinct from #2946, which is a SIGABRT/memory-fault at large shapes on gfx942,
and from #3933, which is a **stage1** `_xcd4` accuracy issue producing finite-but-wrong
values at bucket 4).

## Environment

- aiter commit: `02f6547ba`
- GPU: AMD Instinct **MI350X**, arch **gfx950**
- ROCm: **7.2.0**
- dtype: a4w4 (activation fp4 e2m1, weight fp4 e2m1), scale `per_1x32` (e8m0), bf16 out
- gate: separated, act: silu

## Reproduction

DeepSeek-V3 a4w4 MoE shape: `model_dim=7168, inter_dim=256, E=257, topk=9`, single GPU.

Sweep over token count M (strict correctness, fp32 reference):

```
M=1:  logits_diff = nan    correctness = FAIL
M=2:  nan                  FAIL
M=4:  nan                  FAIL
M=8:  logits_diff = 1.08e-05   PASS
M=16: 1.06e-05                 PASS
M=32: 1.03e-05                 PASS
```

`nan` at M=1 is deterministic across 3 repeats and across input seeds
{0,1,7,42,123,2024}.

## In-path finiteness trace (DS V3, M=1, a4w4, seed 0)

Captured inside the actual two-stage MoE path; every tensor's finiteness checked
once at the end (no mid-stream device syncs — those themselves fault the fp4 path):

```
tensor            finite   absmax        notes
input             True     4.219
score             True     2.625
topk_weights      True     0.170
topk_ids          True     237
a1_qt   (fp4)     True     254           activation fp4 quant (packed bytes finite)
a1_scale (e8m0)   True     1.0
out1_ref          True     45056         torch stage1 reference (finite)
a2_qt   (fp4)     True     249           stage2 activation fp4 quant (finite)
a2_scale (e8m0)   True     8192
out2_ref          True     96768         <-- torch STAGE2 REFERENCE is FINITE
out2_ck           FALSE    3.4028e38     <-- CK STAGE2 KERNEL: nan=6681 inf=487
```

Selected kernel for this shape/bucket:

```
kernelName1 = flydsl_moe1_afp4_wfp4_bf16_t32x128x256_w3_kb14_fp4
kernelName2 = moe_ck2stages_gemm2_64x32x32x128_1x1_MulABScaleExpertWeightShuffled_v1_Nswizzle0_Quant3_MulRoutedWeight1_FP4X2_FP4X2_B16
```

The first and only non-finite tensor is `out2_ck`, the **CK stage2 kernel output**.
The entire torch reference path (including `out2_ref`, absmax 96768) is finite, and
all fp4-quantized inputs/scales (`a1_qt/a1_scale/a2_qt/a2_scale`) are finite. So the
defect is inside the CK stage2 fp4 kernel at small M, not in the reference, the input
quant, or the routing.

## Impact

- a4w4 MoE decode at low concurrency (per-step token count ≤ 4, which maps to the
  affected buckets) produces NaN/Inf logits → garbage tokens.
- It also makes these tiny-M rows uncomparable in any strict correctness gate that
  spans the full token sweep (this is how we hit it: benchmarking a FlyDSL stage2
  dispatch optimization, the DS-V3/Kimi M∈{1,2,4} rows fail the gate purely on the
  CK stage2 kernel output, unrelated to the FlyDSL change).

## What we'd like to understand

1. Is the M ≤ 4 case hitting an out-of-bounds / uninitialized accumulator path in
   the CK stage2 fp4 kernel (the `3.4028e38` absmax strongly suggests an
   uninitialized / overflowed fp32 accumulator rather than a real numeric overflow)?
2. Is `get_padded_M` padding M up to the kernel's expected block_m for these buckets,
   or does block_m=32 with M=1 leave part of the tile reading uninitialized memory?

Happy to provide the full raw trace and the strict-point harness.
